src/lib/homestore_backend/replication_state_machine.cpp

```diff
@@ -2,67 +2,76 @@
     LOGI("applying raft log commit with lsn:{}", lsn);
     const ReplicationMessageHeader* msg_header = r_cast< const ReplicationMessageHeader* >(header.bytes);
-
-    switch (msg_header->message_type) {
-    case ReplicationMessageType::SHARD_MESSAGE: {
-        _home_object->on_shard_message_commit(lsn, header, key, ctx);
+    switch (msg_header->msg_type) {
+    case ReplicationMessageType::CREATE_SHARD_MSG:
+    case ReplicationMessageType::SEAL_SHARD_MSG: {
+        home_object_->on_shard_message_commit(lsn, header, pbas, repl_dev(), ctx);
         break;
     }
-    case ReplicationMessageType::PG_MESSAGE:
-    case ReplicationMessageType::BLOB_MESSAGE:
+    case ReplicationMessageType::PUT_BLOB_MSG:
+    case ReplicationMessageType::DEL_BLOB_MSG:
     default: {
         break;
     }
     }
 }
 
-bool ReplicationStateMachine::on_pre_commit(int64_t lsn, sisl::blob const& header, sisl::blob const& key,
-                                            cintrusive< homestore::repl_req_ctx >& ctx) {
-    bool ret{false};
+bool ReplicationStateMachine::on_pre_commit(int64_t lsn, sisl::blob const&, sisl::blob const&,
+                                            cintrusive< homestore::repl_req_ctx >&) {
     LOGI("on_pre_commit with lsn:{}", lsn);
-    const ReplicationMessageHeader* msg_header = r_cast< const ReplicationMessageHeader* >(header.bytes);
+    // For shard creation, since homestore repldev inside will write shard header to data service first before this
+    // function is called. So there is nothing is needed to do and we can get the binding chunk_id with the newly shard
+    // from the blkid in on_commit()
+    return true;
+}
 
-    switch (msg_header->message_type) {
-    case ReplicationMessageType::SHARD_MESSAGE: {
-        ret = _home_object->on_pre_commit_shard_msg(lsn, header, key, ctx);
-        break;
-    }
-    case ReplicationMessageType::PG_MESSAGE:
-    case ReplicationMessageType::BLOB_MESSAGE:
-    default: {
-        break;
-    }
-    }
-    return ret;
+void ReplicationStateMachine::on_rollback(int64_t lsn, sisl::blob const&, sisl::blob const&,
+                                          cintrusive< homestore::repl_req_ctx >&) {
+    LOGI("on_rollback  with lsn:{}", lsn);
 }
 
-void ReplicationStateMachine::on_rollback(int64_t lsn, sisl::blob const& header, sisl::blob const& key,
-                                          cintrusive< homestore::repl_req_ctx >& ctx) {
-    LOGI("rollback with lsn:{}", lsn);
+homestore::blk_alloc_hints ReplicationStateMachine::get_blk_alloc_hints(sisl::blob const& header,
+                                                                        cintrusive< homestore::repl_req_ctx >& ctx) {
     const ReplicationMessageHeader* msg_header = r_cast< const ReplicationMessageHeader* >(header.bytes);
+    switch (msg_header->msg_type) {
+    case ReplicationMessageType::CREATE_SHARD_MSG: {
+        auto list_shard_result = home_object_->shard_manager()->list_shards(msg_header->pg_id).get();
```

Comment:
can we simplify the logic here a little bit by `home_object_->get_any_chunk_id(msg_header->pg_id)` which in its implementation just call shard_manager to pick *any* shard and return's chunk_id? It seems uncessary to return the full shard lists which can be thousands of shards, right?